### PR TITLE
[monorepo] add travis + codecov ✅

### PR DESCRIPTION
**Pull request by williaster**
This PR attempts to add travis to the build so it fails when prettier, linting, or testing fails.

@kristw @conglei @john-bodley @mistercrunch

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+cache:
+  directories:
+    - node_modules
+
+matrix:
+  fast_finish: true
+
+install:
+  - npm install && lerna bootstrap
```
Comment (john-bodley) [inline .travis.yml]: I believe you can list each command seperately.

File: package.json [hunk 2]
```diff
@@ -27,6 +27,7 @@
   ],
   "license": "Apache-2.0",
   "devDependencies": {
+    "coveralls": "^3.0.2",
```
[the remark this answers is not in the document]
Comment (williaster) [reply package.json]: yes, I initially tried codecov and couldn't add it to this repo so tried coveralls ... still can't add it to the repo so I'll move back to coveralls 👍

[the remark this answers is not in the document]
Comment (john-bodley) [reply package.json]: So the after-script command fails, 
```
$ ./node_modules/.bin/codecov
/home/travis/.travis/job_stages: line 78: ./node_modules/.bin/codecov: No such file or directory
```

**Comment by williaster** [comment]
things are all getting executed correctly, trying to debug why coverage %s are being reported as [unknown](https://travis-ci.com/apache-superset/superset-ui/builds/84184547#L1748).

**Comment by williaster** [comment]
It only took 16 commits but 🎉

File: .travis.yml [hunk 3]
```diff
@@ -20,4 +21,4 @@ script:
   - 'cd ./packages/$PACKAGE && yarn install && yarn run lint && yarn run test'
 
 after_script:
-  - './node_modules/codecov/bin/codecov'
+  - 'codecov'
```
Comment (john-bodley) [inline .travis.yml]: Nix the quotes.

Comment (williaster) [reply .travis.yml]: @john-bodley for all scripts?

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+install:
+  - npm install
+  - npm install -g codecov
+
+env:
+  - PACKAGE=superset-ui-core
+
+script:
+  - cd ./packages/$PACKAGE && yarn install && yarn run lint && yarn run test
```
Comment (john-bodley) [inline .travis.yml]: You can split the script to br a series of commands, 
```
script:
  - cd ./packages/$PACKAGE
  - yarn install
  - ...
```

Comment (williaster) [reply .travis.yml]: that's prob nicer for debugging too 👍

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
+language: node_js
+
+node_js:
+  - '10.7'
```
Comment (john-bodley) [inline .travis.yml]: Nix the quotes here too.

**Comment by williaster** [comment]
gonna merge this into the wrong branch to get a cherry pickable hash for master. don't think I can change the branch I'm merging into now and the other is merged.

**Pull request merged**